src/components/input/input.js

```diff
@@ -362,7 +362,25 @@ function mdMaxlengthDirective($animate) {
     // Stop model from trimming. This makes it so whitespace
     // over the maxlength still counts as invalid.
     attr.$set('ngTrim', 'false');
-    input.after(charCountEl);
+
+    var ngMessagesSelectors = [
+      'ng-messages',
+      'data-ng-messages',
+      'x-ng-messages',
+      '[ng-messages]',
+      '[data-ng-messages]',
+      '[x-ng-messages]'
+    ];
```

Comment:
@jelbourn You mean an `ng-messages` directive that requires the `md-input-container` and registers itself with the container; then this directive just asks the container if there are any messages.

That definitely sounds a bit cleaner; I'll take a look. I think we already have one I can reuse...